website/project/model.py

```diff
@@ -852,15 +855,43 @@ class Node(GuidStoredObject, AddonModelMixin, IdentifierMixin):
     # Primary institution node is attached to
     primary_institution = fields.ForeignField('institution')
 
+    affiliated_institutions = fields.ForeignField('institution', list=True)
+
     def add_primary_institution(self, user, inst):
         if inst.auth(user):
             self.primary_institution = inst
+            self.affiliated_institutions.append(inst)
             self.save()
+            self.add_log(
+                action=NodeLog.INSTITUTION_ADDED,
+                params={
+                    'node': self._primary_key,
+                    'institution': {
+                        'id': inst._id,
+                        'name': inst.name
+                    }
+                },
+                auth=Auth(user)
+            )
             return True
         return False
```

Comment:
I'm not sure returning `False` is the best behavior if the user is not affiliated with the institution. After addressing https://github.com/CenterForOpenScience/osf.io/pull/5017/files#r52918842, there would be no way for the caller to distinguish between

1. Institution not added because it has already been added
2. Institution not added because user is not affiliated with the institution

It would be better if an exception were raised for case 2. Case 1 can probably just return `False`.